api/src/test/java/org/openmrs/validator/DrugOrderValidatorTest.java

```diff
@@ -649,4 +650,118 @@ public void validate_shouldFailIfDurationUnitsHasNoMappingToSNOMEDCTSource() thr
 		assertEquals("DrugOrder.error.durationUnitsNotMappedToSnomedCtDurationCode", errors.getFieldError("durationUnits")
 		        .getCode());
 	}
+	
+	@Test
+	@Verifies(value = "should fail validation if drug non coded is not set for non coded drug order", method = "validate(Object,Errors)")
+	public void saveOrder_shouldFailDrugOrderWithoutADrugNonCodedWhenDrugOrderIsNonCoded() throws Exception {
+		Patient patient = Context.getPatientService().getPatient(7);
+		CareSetting careSetting = Context.getOrderService().getCareSetting(2);
+		OrderType orderType = Context.getOrderService().getOrderTypeByName("Drug order");
+
+		assertNull(Context.getAdministrationService().getGlobalPropertyObject(
```

Comment:
If concept is drugOther, String drugNonCoded is compulsory.. this test is for that validation